### ISSUE TITLE
✨ Support private go dependencies in module development

### ISSUE THREAD

**Issue by nadirollo** [dagger/dagger#8766]
### What are you trying to do?

I am developing a module for internal systems developed in go.

To interact with such systems from my dagger module, I need to import private go dependencies.

There isn't currently a way to provide credentials to the Go runtime container in a way that the codegen will be able to pull such private dependencies (`GOPRIVATE` / modified `gitconfig`)

For a bit more context: https://discord.com/channels/707636530424053791/1298541718374715443

### Why is this important to you?

To be able to develop dagger modules for internal systems that provide libraries to interact with them

**Comment by gtabrar** [comment]
`GOPROXY` support could also be an option for us if it didn't require a custom runner, but as it stands, the UX of getting all our engineers to set that up wouldn't be great. This would definitely be preferable!

**Comment by sipsma** [comment]
> cc @sipsma since you worked on this feature in the first place, maybe you envisioned having an engine-agnostic way of doing it eventually?

Yep, mentioned in the description of the issue here: https://github.com/dagger/dagger/issues/6599

SGTM!

**Comment by jedevc** [comment]
Also x-linking https://github.com/dagger/dagger/issues/7787, which feels like a similar thing in terms of wanting to mount the `.gitconfig` over.

**Comment by nadirollo** [comment]
https://github.com/dagger/dagger/issues/8766#issuecomment-2474813161 > For our use case, the issue https://github.com/dagger/dagger/issues/7787 is likely to implement what we need. 

Being able to pass the GOPROXY would allow us to work around the issue (we could set up a proxy to the private git repositories where our go dependencies are).

The engine able to read the `.gitconfig` with URL rewrites will solve it right away, without having to set up any other component in the middle.
